user.py

```diff
@@ -64,6 +65,9 @@ def add_user():
                                         max_loads=max_load)
             db.session.add(new_researcher)
             db.session.commit()
+    except ValueError as e:
```

Comment:
So here either you make the check before adding the new user to the `db.session`, either you catch an `sqlachemy.exc.IntegrityError` and return that the user already exists in database as your flash message.